lib/Service/SocialApiService.php

```diff
@@ -182,4 +215,82 @@ public function updateContact(string $addressbookId, string $contactId, string $
 		}
 		return new JSONResponse([], Http::STATUS_OK);
 	}
+
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * Updates social profile data for all contacts of an addressbook
+	 * // TODO: how to exclude certain contacts?
+	 *
+	 * @param {String} network the social network to use (fallback: take first match)
+	 * @param {String} user the address book owner
+	 *
+	 * @returns {JSONResponse} JSONResponse with the list of changed and failed contacts
+	 */
+	public function updateAddressbooks(string $network, string $user) : JSONResponse {
```

Comment:
We usually use `userId` when we use the id, $user is usually refering to the IUser class :)